Providers/Profile.swift

```diff
@@ -438,70 +404,52 @@ open class BrowserProfile: Profile {
         }
     }()
 
     /**
-     * Favicons, history, and tabs are all stored in one intermeshed
+     * Favicons, and pinned sites, and are stored in one intermeshed
      * collection of tables.
      *
      * Any other class that needs to access any one of these should ensure
      * that this is initialized first.
      */
-    private lazy var legacyPlaces: Favicons & Profile.HistoryFetcher & HistoryRecommendations  = {
-        return SQLiteHistory(database: self.database, prefs: self.prefs)
+    private lazy var legacyPlaces: Favicons & PinnedSites  = {
+        return BrowserDBSQLite(database: self.database, prefs: self.prefs)
     }()
 
     var favicons: Favicons {
         return self.legacyPlaces
     }
 
-    var history: Profile.HistoryFetcher {
+    var pinnedSites: PinnedSites {
         return self.legacyPlaces
     }
 
     lazy var metadata: Metadata = {
         return SQLiteMetadata(db: self.database)
     }()
 
-    var recommendations: HistoryRecommendations {
-        return self.legacyPlaces
-    }
-
     lazy var placesDbPath = URL(fileURLWithPath: (try! files.getAndEnsureDirectory()), isDirectory: true).appendingPathComponent("places.db").path
     lazy var browserDbPath =  URL(fileURLWithPath: (try! self.files.getAndEnsureDirectory())).appendingPathComponent("browser.db").path
     lazy var places: RustPlaces = RustPlaces(databasePath: self.placesDbPath)
 
-    public func migrateHistoryToPlaces(migrationConfig: HistoryMigrationConfiguration, callback: @escaping (HistoryMigrationResult) -> Void, errCallback: @escaping (Error?) -> Void) {
-        let lastSyncTimestamp = Int64(self.syncManager.lastSyncFinishTime ?? 0)
-        switch migrationConfig {
-        case .disabled:
-            break
-        case .dryRun:
-            let dryRunPath = URL(fileURLWithPath: (try! self.files.getAndEnsureDirectory())).appendingPathComponent("dry-run-places.db").path
-            RustPlaces(databasePath: dryRunPath)
-                .migrateHistory(
-                    dbPath: self.browserDbPath,
-                    lastSyncTimestamp: lastSyncTimestamp,
-                    completion: { result in
-                    do {
-                        try FileManager.default.removeItem(atPath: dryRunPath)
-                    } catch {
-                        log.error("Unable do delete dry run database")
-                    }
-                    callback(result)
-                },
-                errCallback: errCallback)
-        case .real:
-            self.places.migrateHistory(
-                dbPath: self.browserDbPath,
-                lastSyncTimestamp: lastSyncTimestamp,
-                completion: callback,
-                errCallback: errCallback
-            )
+    public func migrateHistoryToPlaces(callback: @escaping (HistoryMigrationResult) -> Void, errCallback: @escaping (Error?) -> Void) {
+        guard FileManager.default.fileExists(atPath: self.browserDbPath) else {
```

Comment:
nit, `self.` can be removed I think
```suggestion
        guard FileManager.default.fileExists(atPath: browserDbPath) else {
```